src/core.c/CompUnit/Repository/Installation.rakumod

```diff
@@ -744,6 +816,239 @@ Please ask the author to add a \"provides\" section, mapping every exposed names
 file location in the distribution.
 See http://design.raku.org/S22.html#provides for more information.\n";
     }
+
+    # Runner generation logic starts here.
+
+    my %arg-op = missing   => 0,
+                 plain     => 1,
+                 cmd-args  => 2,
+                 plat-sep  => 3,
+                 abs       => 4,
+                 abs-slash => 5,
+    ;
+
+    sub validate-env-var($name) {
+        die "$name is not a valid env var name" unless $name ~~ / ^ <[ a..z A..Z _ ]> <[ a..z A..Z 0..9 _ ]>* $ /;
+    }
+
+    sub generate-posix(
+        Str           :$program!,
+        Str           :$cwd,
+                      :@args,
+                      :%env-add,
+                      :@env-remove,
+    ) {
+        my @call-parts = "exec";
+
+        sub sh-escape(Str $text) {
+            S/ ( \$ | \` | \" | \\ | \n ) / \\$1 / given $text;
+        }
+
+        sub process-path(Str() $path) {
+            if $path ~~ /
+                    "<" ( "plain" || "cmd-args" || "plat-sep" || "abs" || "abs-slash" ) ">"
+                    (.*)
+                / {
+                my $op = $0.Str;
+                my $val = $1.Str;
+                given $op {
+                    when "plain" | "plat-sep" {
+                        return '"' ~ sh-escape($val) ~ '"';
+                    }
+                    when "cmd-args" {
+                        return '"$@"';
+                    }
+                    when "abs" | "abs-slash" {
+                        return '"${DIR}/' ~ sh-escape($val) ~ '"';
+                    }
+                }
+            }
+            else {
+                note "Don't understand this arg: $path";
+                exit 1;
```

Comment:
We should probably be returning a failure or `die`ing instead of `exit`ing since this code can be reached by public methods on CURI

Reply:
Done.